src/mono/sample/iOS/Program.cs

```diff
@@ -10,18 +11,32 @@ public static class Program
 {
     // Defined in main.m
     [DllImport("__Internal")]
-    private static extern void ios_set_text(string value);
+    unsafe private static extern void ios_set_text(byte* value);
+    //private static extern void ios_set_text(string value);
```

Comment:
Could we have these under an `#if LIGHTWEIGHT_MARSHALLER` preprocessor directive, which can be controllable from the sample project file via:
```xml
<DefineConstants Condition="'$(UseLightweightMarshallingFriendlyCode)' == 'true'">LIGHTWEIGHT_MARSHALLER;$(DefineConstants)</DefineConstants>
```
This way the sample can still be testable for both implementations (full and lightweight) buy specifying `-p:UseLightweightMarshallingFriendlyCode=true` during build.

NOTE: Maybe the naming I used is not the best

Reply:
This is certainly doable, however I think the point is that the inclusion (or not) of `marhsal-ilgen` should be automatic. So whenever someone tests code in HelloWorld which requires it, it should kick in by by itself. However it could be a good idea to have a test which marshals something nonblittable to check whether the marshaller is correctly included.

Reply:
Maybe we can have such test(s) in the: https://github.com/dotnet/runtime/tree/main/src/tests/FunctionalTests

Reply:
The scenario appears to be covered already, this test https://github.com/dotnet/runtime/blob/main/src/tests/FunctionalTests/iOS/Simulator/AOT-LLVM/Program.cs test should fail if the marshaller is not properly included.